### PR TITLE
Fix the issue with passing the wrong branch type to Flexus.

### PR DESCRIPTION
**Pull request by neo-apz**

### CODE REVIEW

File: target/arm/translate-a64.c [hunk 1]
```diff
@@ -1392,6 +1410,16 @@ static void disas_comp_b_imm(DisasContext *s, uint32_t insn)
 
     gen_goto_tb(s, 0, s->pc);
     gen_set_label(label_match);
+
+#ifdef CONFIG_FLEXUS
+    FLEXUS_IF_IN_SIMULATION( gen_helper_flexus_insn_fetch_aa64( cpu_env,
```
Comment (Hnefi) [inline target/arm/translate-a64.c:1415]: @neo-apz Should this line go before all the gen_goto_tb's? It's worth a test with this specific opcode to make sure that these callbacks fire.

File: target/arm/translate-a64.c [hunk 2]
```diff
@@ -1421,6 +1449,16 @@ static void disas_test_b_imm(DisasContext *s, uint32_t insn)
     tcg_temp_free_i64(tcg_cmp);
     gen_goto_tb(s, 0, s->pc);
     gen_set_label(label_match);
+
+#ifdef CONFIG_FLEXUS
+    FLEXUS_IF_IN_SIMULATION( gen_helper_flexus_insn_fetch_aa64( cpu_env,
```
Comment (Hnefi) [inline target/arm/translate-a64.c:1454]: Same here as above

File: target/arm/translate-a64.c [hunk 3]
```diff
@@ -1448,9 +1486,28 @@ static void disas_cond_b_imm(DisasContext *s, uint32_t insn)
         arm_gen_test_cc(cond, label_match);
         gen_goto_tb(s, 0, s->pc);
         gen_set_label(label_match);
+
+#ifdef CONFIG_FLEXUS
+        FLEXUS_IF_IN_SIMULATION( gen_helper_flexus_insn_fetch_aa64( cpu_env,
+                                        tcg_const_tl(flexus_ins_pc),
```
Comment (Hnefi) [inline target/arm/translate-a64.c:1492]: Same here. I won't spam comment anymore, but i think w can solve this pretty quickly with a test

**Comment by neo-apz** [comment]
The comments are addressed in the new commit.

**Pull request merged**